src/main/java/GradingSystem.java

```diff
@@ -0,0 +1,73 @@
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class GradingSystem {
+    private static final Map<Student, Map<Course, Integer>> studentsCoursesWithGradesMap = new HashMap<>();
```

Comment:
I don't see a reason for this to be a static field

Reply:
You may complicated this class with using` Map<Student, Map<Course, Integer>> ` maybe it would be better if you created a separate class where you would store the user and his grades